cmd/krew/cmd/namingutils_test.go

```diff
@@ -112,3 +112,15 @@ func Test_canonicalName(t *testing.T) {
 		t.Errorf("expected=%q; got=%q", expected, got)
 	}
 }
+
+func Test_isCanonicalName(t *testing.T) {
```

Comment:
while you're at it, some good tests to add:

- ""
- "0-0"
- "a/"
- "/b"
- "a-a/b-b"
- "a//b"
- "a / b"
- "a/ b", "a /b"

and convert it to tabular test format.